extension/amqp/publisher.go

```diff
@@ -0,0 +1,66 @@
+package amqp
+
+import (
+	"context"
+	"github.com/hellofresh/goengine"
+	"github.com/hellofresh/goengine/driver/sql"
+	"github.com/mailru/easyjson"
+	"github.com/streadway/amqp"
+)
+
+var _ sql.ProjectionTrigger = (&NotificationPublisher{}).Publish
+
+type NotificationPublisher struct {
+	amqpDSN string
+	queue   string
+	logger  goengine.Logger
+
+	connection *amqp.Connection
+	channel    *amqp.Channel
+}
+
+func NewNotificationPublisher(amqpDSN, queue string, logger goengine.Logger) (*NotificationPublisher, error) {
+	return &NotificationPublisher{
+		amqpDSN: amqpDSN,
+		queue:   queue,
+		logger:  logger,
+	}, nil
+}
+
+func (p *NotificationPublisher) Publish(ctx context.Context, notification *sql.ProjectionNotification) error {
+	// Ignore nil notifications since this is not supported
+	if notification == nil {
+		p.logger.Warn("unable to handle nil notification, skipping", nil)
+		return nil
+	}
+
+	msgBody, err := easyjson.Marshal(notification)
+	if err != nil {
+		return err
+	}
+
+	for {
+		if p.connection == nil {
+			p.connection, p.channel, err = setup(p.amqpDSN, p.queue)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = p.channel.Publish("", p.queue, true, false, amqp.Publishing{
```

Comment:
@boekkooi-fresh , does it make sense to use `NotifyPublish` and confirm message delivery here? As our messages are already persistent in event store and will be picked again in case of failure or so. I believe there's no need to do that. wdyt?

Reply:
I think that for our specific use case this would be ok but since we are dealing with an opensource library I personally feel that we should do it the correct way :smile: